### PR TITLE
ReadTheDocs Configuration File

### PR DESCRIPTION
**Pull request by amandarichardsonn**
Adding readthedocs yaml file to develop

### CODE REVIEW

File: doc/conf.py
Comment (ashao) [inline doc/conf.py:86]: This will create a robots.txt file for all documentation that we build (including the one that will be hosted at craylabs.org. I was digging into the documentation a little more (and verified) that readthedocs by default ships a robots.txt: https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html#seo-robots-txt

This seems to be the case for us, since this file exists: https://smartsim.readthedocs.io/en/latest/robots.txt

Could you dig into this a little more? It might also (for safety) be worthwhile to still figure out a way to make a robots.txt file but have a way to only create the robots.txt file if we know that it's being built via readthedocs

Could you dig into this and

**Review by ashao — CHANGES_REQUESTED** [review]
One comment on the robots.txt work

Another item to take care of requires configuring ReadTheDocs:
- Make craylabs.org the canonical URL: https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html#canonical-urls
- Add me, Matt Ellis, and Al as owners of the SmartSim readthedocs account

File: doc/conf.py
Comment (amandarichardsonn) [reply doc/conf.py:86]: for sure!

**Comment by amandarichardsonn** [comment]
canonical URL added! You can check it [here](https://readthedocs.org/dashboard/smartsim/domains/) (Details).

File: doc/changelog.rst
Comment (mellis13) [inline doc/changelog.rst:29]: Thanks for fixing the changelog.  Can you also fix the order in the detailed description section.  The last two items are out-of-order compared to the top short descriptions.

File: doc/changelog.rst
Comment (mellis13) [inline doc/changelog.rst:28]: Also, I think we need a changelog entry for this PR

Comment (amandarichardsonn) [reply doc/changelog.rst:32]: updated!

File: doc/changelog.rst
Comment (amandarichardsonn) [reply doc/changelog.rst:29]: fixed

File: .readthedocs.yaml
Comment (mellis13) [inline .readthedocs.yaml:11]: changed -> changes

File: .readthedocs.yaml
Comment (mellis13) [inline .readthedocs.yaml:13]: like your docs requirements file, or other files that will change your docs build.

->

such as docs requirements file or other files that affect the docs build.

File: .readthedocs.yaml
Comment (mellis13) [inline .readthedocs.yaml:15]: Great comment!  Thanks for explaining this.

**Review by mellis13 — APPROVED** [review]
Approving with only minor comments and pending @ashao review

File: .readthedocs.yaml
Comment (ashao) [inline .readthedocs.yaml:32]: Could you make a ticket to update the `make docs` target so that we can eventually converge the two workflows?

File: doc/conf.py
Comment (ashao) [inline doc/conf.py:88]: Nice solution!

File: doc/conf.py
Comment (ashao) [inline doc/conf.py:90]: Why is this `../` and not `./`?

**Review by ashao — APPROVED** [review]
Thanks for your work on this! Some very minor comments that shouldn't hold up merging.

In addition to the `make docs` update ticket that I suggest, I'm going to open up another one about the canonical URL. What it looks like that means is actually that we could update craylabs.org to point to ReadTheDocs instead. That's not happening now, because we don't have our domain provider set up like that, but I think we should think through whether we want to do that.

File: .readthedocs.yaml
Comment (amandarichardsonn) [reply .readthedocs.yaml:32]: yes!

**Pull request merged**
